tests/ui/methods.stderr

```diff
@@ -103,65 +103,78 @@ error: unnecessary structure name repetition
-   --> $DIR/methods.rs:103:13
+   --> $DIR/methods.rs:103:16
     |
 103 |       let _ = opt.map(|x| x + 1)
-    |  _____________^
+    |  ________________^
 104 | |
 105 | |                .unwrap_or(0); // should lint even though this call is on a separate line
-    | |____________________________^
+    | |____________________________^ help: try this: `.map_or(0, |x| x + 1)`
     |
     = note: `-D option-map-unwrap-or` implied by `-D warnings`
-    = note: replace `map(|x| x + 1).unwrap_or(0)` with `map_or(0, |x| x + 1)`
 
 error: called `map(f).unwrap_or(a)` on an Option value. This can be done more directly by calling `map_or(a, f)` instead
-   --> $DIR/methods.rs:107:13
+   --> $DIR/methods.rs:107:16
     |
 107 |       let _ = opt.map(|x| {
-    |  _____________^
+    |  ________________^
 108 | |                         x + 1
 109 | |                     }
 110 | |               ).unwrap_or(0);
     | |____________________________^
+help: try this
+    |
+107 |     let _ = opt.map_or(0, |x| {
+108 |                         x + 1
+109 |                     });
+    |
 
 error: called `map(f).unwrap_or(a)` on an Option value. This can be done more directly by calling `map_or(a, f)` instead
-   --> $DIR/methods.rs:111:13
+   --> $DIR/methods.rs:111:16
     |
 111 |       let _ = opt.map(|x| x + 1)
-    |  _____________^
+    |  ________________^
 112 | |                .unwrap_or({
 113 | |                     0
 114 | |                 });
     | |__________________^
+help: try this
+    |
+111 |     let _ = opt.map_or({
+112 |                     0
+113 |                 }, |x| x + 1);
+    |
 
 error: called `map(f).unwrap_or(None)` on an Option value. This can be done more directly by calling `and_then(f)` instead
-   --> $DIR/methods.rs:116:13
+   --> $DIR/methods.rs:116:16
     |
 116 |     let _ = opt.map(|x| Some(x + 1)).unwrap_or(None);
-    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-    |
-    = note: replace `map(|x| Some(x + 1)).unwrap_or(None)` with `and_then(|x| Some(x + 1))`
+    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `.and_then(|x| Some(x + 1))`
 
 error: called `map(f).unwrap_or(None)` on an Option value. This can be done more directly by calling `and_then(f)` instead
-   --> $DIR/methods.rs:118:13
+   --> $DIR/methods.rs:118:16
     |
 118 |       let _ = opt.map(|x| {
-    |  _____________^
+    |  ________________^
 119 | |         Some(x + 1)
 120 | |     }
 121 | |     ).unwrap_or(None);
     | |_____________________^
+help: try this
+    |
+118 |     let _ = opt.and_then(|x| {
+119 |         Some(x + 1)
+120 |     });
+    |
 
 error: called `map(f).unwrap_or(None)` on an Option value. This can be done more directly by calling `and_then(f)` instead
-   --> $DIR/methods.rs:122:13
+   --> $DIR/methods.rs:122:16
     |
 122 |       let _ = opt
-    |  _____________^
+    |  ________________^
```

Comment:
…and this looks very weird

Reply:
multi line spans are weird. Would be solved by the above suggestion, too